Network Impact Showcase

lucascumsille opened the pull request:
## Relevant issue(s)

Fixes: https://github.com/mysociety/alaveteli/issues/8928

## What does this do?

Create the Network Impact Showcase page 

## Why was this needed?

## Implementation notes

TODOs

- [x] Delete placeholder images
- [x] Delete fake cards
- [x] Get rid of Lorem ipsum quotes
- [x] Style "Load more" button
- [x] Continue adding the rest [23 cards left ](https://docs.google.com/document/d/1TO5OJxuyB7wtr5NfW-bZXV_74mbVtONFtLemtmRilbs/edit?tab=t.0)
- [x] After testing those 23 cards change `masonry-initial-cards="100"` to something like 20.
- [x] We need to decide which SCSS variable we will use to theme this page. At the moment we are using`$color_primary` but that one does not exist in `whatdotheyknow theme` and `none` theme. `button-bg` might be a good candidate for `whatdotheyknow`, but not so much for `alaveteli` because `button_bg` is using `color_green` which doesn't match the purple theme that alaveteli is known for. We could also create a variable `color_primary_showcase` that variable could feed all the rest when it comes to colour. After speaking with Gareth I think we will move ahead with `color_primary_showcase`, but we need a fallback, so if the collaborator has not declared the variable then it uses a default, that might look bland, but it is easy to change if they want to. We should update the changelog if we end up doing this.
- [x] In the Hero section change order the sites. Do alphabetically by country.
- [x] The AI card (How access to information can help us understand AI decision making) and Tackling the climate crisis with Access To Information rights should use `width2`
- [x] Increase the width of the modal
- [x] Style the button of the modal.
- [x] Get rid of the gap before the footer.
- [x] Add a quote section, like we do in FMS Pro.
- [x] The Board section should also have quotes. This quotes should NOT look like cards, so get rid of the box-shadows and borders.
- [x] The homepage will need a CTA. Gareth will mock up something.
  - https://github.com/mysociety/alaveteli/issues/8928#issuecomment-3580933220
- [x] Fix deprecation warnings
- [x] Add a bit of text at the end of the cards to ask readers to tell the site about more examples (link to `help_contact_path`)

## Screenshots

Version:  26 November 2025 10:14am

https://github.com/user-attachments/assets/037f145a-d5c9-4fd5-9ac2-48aae14b3722


## Notes to reviewer

<hr>

Have you updated the changelog? If this is not necessary, put square brackets around this: [skip changelog]

garethrees reviewed (changes requested):
This doesn't work with the `none` theme or `whatdotheyknow-theme` at the moment.

Looks like its failing because `$color_primary` sass variable doesn't exist in either of those.

```
Started GET "/access-to-information-around-the-world" for 172.24.0.1 at 2025-11-24 14:30:55 +0000
Cannot render console from 172.24.0.1! Allowed networks: 10.10.0.0/10.10.255.255, 127.0.0.0/127.255.255.255, ::1
   (0.9ms)  SELECT "flipper_features"."key" AS feature_key, "flipper_gates"."key", "flipper_gates"."value" FROM "flipper_features" LEFT OUTER JOIN "flipper_gates" ON "flipper_features"."key" = "flipper_gates"."feature_key" /*application='Alaveteli'*/
Processing by GeneralController#access_to_information_around_the_world as HTML
  Rendering layout layouts/default.html.erb
  Rendering general/access_to_information_around_the_world.html.erb within layouts/default
  Rendered general/access_to_information_around_the_world.html.erb within layouts/default (Duration: 1.0ms | GC: 0.0ms)
  Rendered layouts/_favicon.html.erb (Duration: 3051.6ms | GC: 29.9ms)
  Rendered layout layouts/default.html.erb (Duration: 3056.6ms | GC: 29.9ms)
Completed 500 Internal Server Error in 3063ms (ActiveRecord: 0.0ms (0 queries, 0 cached) | GC: 30.0ms)



ActionView::Template::Error (Undefined variable: "$color_primary".)
Caused by: Sass::SyntaxError (Undefined variable: "$color_primary".)

Information for: ActionView::Template::Error (Undefined variable: "$color_primary".):
    1: <%= favicon_link_tag 'favicon.ico' %>

app/views/layouts/_favicon.html.erb:1
app/views/layouts/default.html.erb:15
app/controllers/application_controller.rb:132:in `record_memory'

Information for cause: Sass::SyntaxError (Undefined variable: "$color_primary".):

app/views/layouts/_favicon.html.erb:1
app/views/layouts/default.html.erb:15
app/controllers/application_controller.rb:132:in `record_memory'
```

lucascumsille commented:
> This doesn't work with the `none` theme or `whatdotheyknow-theme` at the moment.
> 
> Looks like its failing because `$color_primary` sass variable doesn't exist in either of those.
> 
> ```
> Started GET "/access-to-information-around-the-world" for 172.24.0.1 at 2025-11-24 14:30:55 +0000
> Cannot render console from 172.24.0.1! Allowed networks: 10.10.0.0/10.10.255.255, 127.0.0.0/127.255.255.255, ::1
>    (0.9ms)  SELECT "flipper_features"."key" AS feature_key, "flipper_gates"."key", "flipper_gates"."value" FROM "flipper_features" LEFT OUTER JOIN "flipper_gates" ON "flipper_features"."key" = "flipper_gates"."feature_key" /*application='Alaveteli'*/
> Processing by GeneralController#access_to_information_around_the_world as HTML
>   Rendering layout layouts/default.html.erb
>   Rendering general/access_to_information_around_the_world.html.erb within layouts/default
>   Rendered general/access_to_information_around_the_world.html.erb within layouts/default (Duration: 1.0ms | GC: 0.0ms)
>   Rendered layouts/_favicon.html.erb (Duration: 3051.6ms | GC: 29.9ms)
>   Rendered layout layouts/default.html.erb (Duration: 3056.6ms | GC: 29.9ms)
> Completed 500 Internal Server Error in 3063ms (ActiveRecord: 0.0ms (0 queries, 0 cached) | GC: 30.0ms)
> 
> 
> 
> ActionView::Template::Error (Undefined variable: "$color_primary".)
> Caused by: Sass::SyntaxError (Undefined variable: "$color_primary".)
> 
> Information for: ActionView::Template::Error (Undefined variable: "$color_primary".):
>     1: <%= favicon_link_tag 'favicon.ico' %>
> 
> app/views/layouts/_favicon.html.erb:1
> app/views/layouts/default.html.erb:15
> app/controllers/application_controller.rb:132:in `record_memory'
> 
> Information for cause: Sass::SyntaxError (Undefined variable: "$color_primary".):
> 
> app/views/layouts/_favicon.html.erb:1
> app/views/layouts/default.html.erb:15
> app/controllers/application_controller.rb:132:in `record_memory'
> ```

mmm not really sure what variable we should use. `$main_menu-bg`? but I can't see that one on the `alavetelitheme`. another one that I thought at the beginning was `button_bg` but that one is green on  `alavetelitheme`. I think we can create with SCSS a conditional if the variable does not exist then use another one.

lucascumsille commented on app/views/general/_responsive_header.html.erb:
```diff
@@ -50,3 +50,9 @@
   <%= render :partial => "frontpage_hero" %>
 <% end %>
+
+<% if params[:action] == 'access_to_information_around_the_world' %>
```
Because they would be wrapped around `#wrapper` and `#content`, the first one has `max-width` the second one horizontal padding. That makes the page looks like this:

https://github.com/user-attachments/assets/0164bde4-0251-49e9-b630-5bfe0a88836d

I added a fixup where I added them to `app/views/general/access_to_information_around_the_world.html.erb` and add rule to `#wrapper` and `#content` so it doesn't have a `max-width` and the padding.

lucascumsille commented:
> Could we make the whole card clickable, rather than just the "Read more" button? If it's a modal, clicking should open the modal. If it's a link, then it should open the link.

I just added a commit fixup for that.

garethrees commented:
Should add some sort of text after we've loaded all the cards to remind people it's just a selection and that the site would love to hear about more examples

garethrees commented:
needs a `<title>`

garethrees commented on app/views/ati_network/showcase/_community_of_practice.html.erb:
```diff
@@ -0,0 +1,26 @@
+      </div>
+    </div>
+
+    <a class="link_button_green" href="https://tictec.mysociety.org/ati-community-of-practice/" class>Join our Access to Information Community of Practice</a>
```
unused class

garethrees commented on app/views/ati_network/showcase/_impact_board.html.erb:
```diff
@@ -0,0 +1,551 @@
+      <p class="card--text">Investigation series around political funding that is hidden or hard to trace that's made extensive use of FOI.</p>
+      <a href="https://www.opendemocracy.net/en/dark-money-investigations/" class="card--button">Read more</a>
+    </div>
+
```
Remove space

garethrees commented on app/views/ati_network/showcase/_impact_board.html.erb:
move to assets

garethrees commented on app/views/layouts/default.html.erb:
```diff
@@ -45,7 +45,7 @@
     <%= content_for :javascript_head %>
   </head>
-  <body class="<%= 'front' if params[:action] == 'frontpage' %> <% if @in_pro_area %>alaveteli-pro<% end %>">
+  <body class="<%= 'front' if params[:action] == 'frontpage' %> <%= 'ati_network_showcase' if params[:action] == 'showcase' %> <% if @in_pro_area %>alaveteli-pro<% end %>">
```
Would be good to avoid needing this in this view. IIRC this was to do with the wrapper?

Could this be moved to `app/views/ati_network/showcase.html.erb` somehow? If not then I think we can do something more generic on the Rails side. I'd have to look at specifics but I'm imagining something like:

```erb
<!-- layout -->
<body class="… <%= yield :body_class %>"

<!-- view -->
<% content_for :body_class { 'no-wrap' } %>
<div>
  …
```

garethrees commented on app/views/ati_network/showcase/_hero.html.erb:
```diff
@@ -0,0 +1,62 @@
+    </div>
+
+    <div class="row hero-list-sites-section">
+      <h2>This site and many others are powered by Alaveteli, a service by <%= image_tag('ati_network/showcase/logo-mysociety.png', alt: _('mySociety'), :class => "hero-mysoc-logo") %></h2>
```
Can we:

- [ ] Make the logo a link to https://www.mysociety.org/
- [ ] Make it inline with the text rather than block, and about the same size as the text so that it looks like its part of the sentence rather than a separate element

lucascumsille commented:
@garethrees I'll leave this here instead of Slack, in case Graeme needs to work on this.

The solution was a bit simple and `!default` variables did the trick.

A couple of notes in order to make it work on your local, see what I did here:

- https://github.com/mysociety/alavetelitheme/pull/122
- https://github.com/mysociety/whatdotheyknow-theme/pull/2069

-------

### None theme

https://github.com/user-attachments/assets/1bee7c65-e28f-4c87-bb2c-353d70128ab7

### WhatDoTheyKnow

https://github.com/user-attachments/assets/5218baae-0eb3-4d2f-a77d-c52d3b05cab8


### Alaveteli Theme
https://github.com/user-attachments/assets/257c43d5-fe0c-410d-b5a5-c492621b2e74

garethrees commented:
another report to add (such a great cover!) https://multinationales.org/IMG/pdf/pfuesousinfluence.pdf

garethrees commented on app/views/ati_network/showcase/_impact_board.html.erb:
```diff
@@ -0,0 +1,496 @@
+    </a>
+
+    <div class="gallery-grid--card">
+      <div class="card--header-badge"><span class="badge--flag">🇫🇷</span>France</div>
```
Somehow this is showing france here (in the diff) but europe on the actual site

<img width="366" height="326" alt="Image" src="https://github.com/user-attachments/assets/d866052c-18c3-424e-82d9-a7e08d75ed93" />

lucascumsille commented:
@garethrees I just push the last commit =)

Here are the previews:

### Homepage

https://github.com/user-attachments/assets/d4480ef2-625c-477a-9ea1-4b3b0803e20f

### ATI Alaveteli

https://github.com/user-attachments/assets/c010ce29-999a-4bee-8944-eb0275173f7a

### ATI WDTK

https://github.com/user-attachments/assets/cbf11654-d890-46f4-91e1-4dbc25349c8f

### ATI No Theme

https://github.com/user-attachments/assets/9a6cc77b-9d0c-4a8e-9a16-d19b6f72bf9d

I haven't rebased so you can see the changes, but here is a summary:

- Replaced CSS variables with SCSS ones.
- Added a variables file so the themes can override the default ones
- Improved comments so it's easier to understand when you want to theme this page.
- Improved hero section with the `mix-blend-mode`
- Deleted the `opacity` in the world-map svg, so we now handle that as a SCSS variable instead.
- Changed the quote section colour, so it also relies in the `color_primary` so we don't have a jarring color palette.
- I decreased the `hue-rotation` values for most of the backgrounds. Same reason as above.
- Updated the `id` and `aria-labelledby` attributes so they don't repeat. 

Let me know if we should change anything.
